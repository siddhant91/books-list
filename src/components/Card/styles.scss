@import '../../styles/main.scss';
.bokl-card {
	background: $bokl-color-white;
	border-radius: $bokl-spacing-2;
	box-shadow: 0px 1px 6px 0px rgba($bokl-color-black, 0.1);
	border-left: $bokl-spacing-2 solid $bokl-color-orange;
}
