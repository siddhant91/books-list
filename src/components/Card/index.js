import clsx from 'clsx';
import { node, string } from 'prop-types';

// styles
import './styles.scss';

function Card({ additionalClasses, children, ...rest }) {
	return (
		<div className={clsx('bokl-card', additionalClasses)} {...rest}>
			{children}
		</div>
	);
}

Card.propTypes = {
	additionalClasses: string,
	children: node,
};
Card.defaultProps = {
	children: null,
	additionalClasses: '',
};

export default Card;
