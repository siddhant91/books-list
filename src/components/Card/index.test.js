import { render } from '@testing-library/react';
import Card from './index';

it('Matches snapshot', () => {
	const { asFragment } = render(<Card> this is a Card</Card>);
	expect(asFragment()).toMatchSnapshot();
});

it('Renders children props', () => {
	const { getByText } = render(
		<Card>
			<p>Test Card</p>
		</Card>,
	);
	expect(getByText('Test Card')).toBeDefined();
});
