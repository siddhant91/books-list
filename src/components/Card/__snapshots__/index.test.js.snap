// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bokl-card"
  >
     this is a Card
  </div>
</DocumentFragment>
`;
